File: Inventory_Backend/src/main/java/com/Inventory/Inventory_Backend/stock/service/StockService.java
package com.Inventory.Inventory_Backend.stock.service;

import com.Inventory.Inventory_Backend.stock.dto.StockAdjustmentRequestDTO;
import com.Inventory.Inventory_Backend.stock.dto.StockMovementResponseDTO;
import com.Inventory.Inventory_Backend.stock.dto.StockResponseDTO;

import java.math.BigDecimal;
import java.util.List;

public interface StockService {

    // =========================================================
    // INCREASE STOCK (PURCHASE)
    // =========================================================
    void increaseStock(Long businessId, Long itemId, BigDecimal quantity, Long referenceId);

    // =========================================================
    // DECREASE STOCK (SALE)
    // =========================================================
    void decreaseStock(Long businessId, Long itemId, BigDecimal quantity, Long referenceId);

    // =========================================================
    // DECREASE STOCK FROM QUOTATION CONVERSION
    // =========================================================
    void decreaseStockFromQuotation(Long businessId, Long itemId, BigDecimal quantity, Long quotationId);

    // =========================================================
    // MANUAL STOCK ADJUSTMENT
    // =========================================================
    void adjustStock(Long businessId, StockAdjustmentRequestDTO request);

    // =========================================================
    // GET STOCK FOR SINGLE ITEM
    // =========================================================
    StockResponseDTO getStock(Long businessId, Long itemId);

    // =========================================================
    // GET ALL STOCK FOR BUSINESS
    // =========================================================
    List<StockResponseDTO> getAllStock(Long businessId);

    // =========================================================
    // GET LOW STOCK ITEMS ⭐
    // =========================================================
    List<StockResponseDTO> getLowStockItems(Long businessId);

    // =========================================================
    // GET STOCK MOVEMENT HISTORY
    // =========================================================
    List<StockMovementResponseDTO> getStockMovements(Long businessId, Long itemId);
}

File: frontend/src/services/StockService.js
import api from "./api";

// ======================================
// GET ALL STOCK
// ======================================
export const getAllStock = async () => {
  try {
    const response = await api.get("/stock");
    return response.data;
  } catch (error) {
    console.error("Error fetching all stock:", error);
    throw error;
  }
};

// ======================================
// GET STOCK BY ITEM ID
// ======================================
export const getStockByItem = async (itemId) => {
  try {
    const response = await api.get(`/stock/${itemId}`);
    return response.data;
  } catch (error) {
    console.error(`Error fetching stock for item ${itemId}:`, error);
    throw error;
  }
};

// ======================================
// GET LOW STOCK ITEMS
// ======================================
export const getLowStockItems = async () => {
  try {
    const response = await api.get("/stock/low-stock");
    return response.data;
  } catch (error) {
    console.error("Error fetching low stock items:", error);
    throw error;
  }
};

// ======================================
// ADJUST STOCK (CREATE / UPDATE)
// ======================================
export const adjustStock = async (payload) => {
  try {
    const response = await api.post("/stock/adjust", payload);
    return response.data;
  } catch (error) {
    console.error("Error adjusting stock:", error);
    throw error;
  }
};

// ======================================
// GET STOCK MOVEMENTS
// ======================================
export const getStockMovements = async (itemId) => {
  try {
    const response = await api.get(`/stock/${itemId}/movements`);
    return response.data;
  } catch (error) {
    console.error(`Error fetching movements for item ${itemId}:`, error);
    throw error;
  }
};

